cli/command/formatter/sorter.go

```diff
@@ -0,0 +1,289 @@
+package formatter
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+)
+
+type genericStructSorter struct {
+	data     []interface{}
+	bys      []string
+	ordering []string
+	indexes  [][]int
+}
+
+func (s genericStructSorter) Len() int {
+	return len(s.data)
+}
+
+func (s genericStructSorter) Swap(i, j int) {
+	s.data[i], s.data[j] = s.data[j], s.data[i]
+}
+
+const (
+	ascendantKey  = "asc"
+	descendantKey = "dsc"
+)
+
+func validateType(i interface{}, j interface{}) error {
+	iType := reflect.TypeOf(i)
+	jType := reflect.TypeOf(j)
+	if iType.Kind() != reflect.Struct {
+		return fmt.Errorf("data must be of type struct")
+	}
+	if iType.PkgPath() != jType.PkgPath() || iType.Name() != jType.Name() || iType.Kind() != jType.Kind() {
+		return fmt.Errorf("data are not of the same type: (%v, %v, %v) != (%v, %v, %v)",
+			iType.PkgPath(), iType.Name(), iType.Kind(), jType.PkgPath(), jType.Name(), jType.Kind())
+	}
+	return nil
+}
+
+func findFieldIndex(t reflect.Type, name string, index *[]int) bool {
+	match := func(s string) bool {
+		return strings.EqualFold(s, name)
+	}
+	f, ok := t.FieldByNameFunc(match)
+	if ok {
+		*index = append(f.Index, *index...)
+		return true
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		nt := t.Field(i).Type
+		if t.Kind() == reflect.Ptr {
+			nt = nt.Elem()
+		}
+		if nt.Kind() == reflect.Struct {
+			ok = findFieldIndex(nt, name, index)
```

Comment:
This seems a bit much. Our list views should be simple, i.e. a struct with simple fields.